### PR TITLE
v1.13 Backports 2023-07-19

### PR DESCRIPTION
**Pull request by gandro**
* [x] #26874 (@michi-covalent)
 * [x] #26812 (@chancez)
 * [x] #26883 (@michi-covalent) :warning: Minor conflicts
 * [x] #26797 (@toVersus)
 * [x] #26856 (@brb)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 26874 26812 26883 26797 26856; do contrib/backporting/set-labels.py $pr done 1.13; done
```
or with
```
make add-labels BRANCH=v1.13 ISSUES=26874,26812,26883,26797,26856
```

### CODE REVIEW

**Comment by gandro** [comment]
/test-backport-1.13

File: .github/workflows/conformance-e2e.yaml [hunk 1]
```diff
@@ -341,7 +341,12 @@ jobs:
           provision: 'false'
           cmd: |
             cd /host/
-            ./contrib/scripts/kind.sh --xdp "" 3 "" "" "${{ matrix.kube-proxy }}" "dual"
+
+            IP_FAM="dual"
+            if [ "${{ matrix.ipv6 }}" == "false" ]; then
+              IP_FAM="ipv4"
+            fi
+            ./contrib/scripts/kind.sh --xdp "" 3 "" "" "${{ matrix.kube-proxy }}" \$IP_FAM
```
Comment (viktor-kurchenko) [inline .github/workflows/conformance-e2e.yaml:349]: Does `\` required here?

**Review by viktor-kurchenko — APPROVED** [review]
LGTM.

File: .github/workflows/conformance-e2e.yaml (hunk 1 above)
Comment (gandro) [reply .github/workflows/conformance-e2e.yaml:349]: This is also present in the upstream commit: https://github.com/cilium/cilium/pull/26856/files @brb 

If this is incorrect, we a fix on `main` first and then backport it.

**Comment by gandro** [comment]
- test-1.26-net-next failed all over the place. I'm not exactly sure why, as previous runs on the branch were fine and there are really no significant changes to the agent. Inspecting the logs also doesn't yield anything of note (e.g. such as a complexity issue). I'm restarting

**Comment by gandro** [comment]
/test-1.26-net-next

**Review by michi-covalent — APPROVED** [review]
wonderful

File: .github/workflows/conformance-e2e.yaml (hunk 1 above)
Comment (brb) [reply .github/workflows/conformance-e2e.yaml:349]: Yep, it's required - https://github.com/cilium/little-vm-helper/blob/main/action.yaml#L21.

**Review by brb — APPROVED** [review]
Thanks!

**Review by chancez — APPROVED** [review]

**Pull request merged**